link/src/main/java/com/stripe/android/link/ui/signup/SignUpViewModel.kt

```diff
@@ -40,18 +45,18 @@ internal class SignUpViewModel @Inject internal constructor(
     /**
      * Emits the email entered in the form if valid, null otherwise.
      */
-    private val consumerEmail: Flow<String?> = emailElement.getFormFieldValueFlow().map {
+    private val consumerEmail: StateFlow<String?> = emailElement.getFormFieldValueFlow().map {
```

Comment:
Can you add a comment to why is it firstOrNull()?.second?
Also I think the logic inside the let can be simplified to: formFieldEntry.value.takeIf { formFieldEntry.isComplete }

Reply:
emailElement.getFormFieldValueFlow() returns a __ list of formFieldEntries, there is only one item in the list, so firstOrNull is used.  Second is because we are looking for the FormFieldValue for the entry.

I do wonder if getFormFieldValueFlow.first().filter{it == email}.second?.takeIf{...}... make it more clear that you are getting just the email field?